src/actions/channels.js

```diff
@@ -827,17 +843,34 @@ export function markChannelAsRead(channelId, prevChannelId) {
             }
         }
 
+        if (teamMember) {
+            actions.push({
+                type: TeamTypes.RECEIVED_MY_TEAM_UNREADS,
+                data: [teamMember]
+            });
+        }
+
         if (actions.length) {
             dispatch(batchActions([...actions]), getState);
         }
     };
 }
 
-export function markChannelAsUnread(channelId, mentionsArray) {
+export function markChannelAsUnread(teamId, channelId, mentionsArray) {
```

Comment:
The problem with that is you may get a post from a channel in a team that hasn't been loaded yet and then you don't have the channel in the store, no channel in the store means no team_id, sounds about right?